### PR TITLE
Enables cross-package code coverage reporting

### PR DESCRIPTION
**Pull request by BrianBland**
Signed-off-by: Brian Bland <brian.bland@docker.com>

### CODE REVIEW

**Comment by dmp42** [comment]
Something is wrong :)

**Comment by BrianBland** [comment]
Godep path... :(

**Comment by dmp42** [comment]
Time to switch to `vendor` and `GO15VENDOREXPERIMENT=1`?

**Comment by BrianBland** [comment]
OK I think it finally works without ruining test performance :+1:

**Comment by dmp42** [comment]
LGTM

File: circle.yml [hunk 1]
```diff
@@ -75,25 +72,17 @@ test:
...
-
   # Test stable, and report
-  # Preset the goverall report file
-  # - echo "$CIRCLE_PAIN" > ~/goverage.report
-
-     - gvm use stable; go list ./... | xargs -L 1 -I{} rm -f $GOPATH/src/{}/coverage.out:
-         pwd: $BASE_STABLE
-
-     - gvm use stable; go list -tags "$DOCKER_BUILDTAGS" ./... | xargs -L 1 -I{} godep go test -tags "$DOCKER_BUILDTAGS" -test.short -coverprofile=$GOPATH/src/{}/coverage.out -covermode=count {}:
+     - gvm use stable; export ROOT_PACKAGE=$(go list .); go list -tags "$DOCKER_BUILDTAGS" ./... | xargs -L 1 -I{} bash -c 'export PACKAGE={}; godep go test -tags "$DOCKER_BUILDTAGS" -test.short -coverprofile=$GOPATH/src/$PACKAGE/coverage.out -coverpkg=$(./coverpkg.sh $PACKAGE $ROOT_PACKAGE) $PACKAGE':
```
Comment (stevvooe) [inline circle.yml]: `GOPATH` is not guaranteed to be a single component. This may do the wrong thing.

**Pull request merged**